[docs] Update the contribution guide (#2553)
DevExpress/testcafe-hammerhead: CONTRIBUTING.md

## Reporting a Problem

> If you run into a bug with TestCafe Hammerhead, please file an issue in the [GitHub repository](https://github.com/DevExpress/testcafe-hammerhead/issues).

Reviewer: ```suggestion
If you encounter a bug with TestCafe Hammerhead, please file an issue in the [GitHub repository](https://github.com/DevExpress/testcafe-hammerhead/issues).
```

Search through the existing issues to see if the problem has already been reported or addressed.

> When you create a new issue, the template text is automatically added to its body. You should complete all the sections in this template to help us understand the issue you are describing. Missing information could delay the processing time.

Reviewer: ```suggestion
When you create a new issue, the template text is automatically added to its body. You should complete all sections in this template to help us understand the issue you are describing. Missing information could delay the processing time.
```

1. Search the [list of issues](https://github.com/DevExpress/testcafe-hammerhead/issues) to see if there is an issue for the bug or feature you are going to work on or create a new one.

> 2. To address an already described issue, check the comment thread to make sure that nobody is working on it at the moment. Leave a comment saying that you are willing to fix this issue, and include details on how you plan to do this. Core team members may need to discuss the details of the proposed fix with you. After the green light from them,

Reviewer: ```suggestion
2. To address an already described issue, check the comment thread to make sure that no one is working on it at the moment. Leave a comment saying that you are willing to fix this issue, and include details on how you plan to do this. Core team members may need to discuss the details of the proposed fix with you. After they have approved it,
```

> leave a comment saying that you started your work on this issue.

Reviewer: Just out of curiousity, why is the last part of line 33 on this line instead?

Author: That's a mistake. Good that you've noticed, thanks.

3. Install [Node.js](https://nodejs.org/en/) on your development machine.

4. Fork TestCafe Hammerhead and create a branch in your fork. Name this branch with an issue number, for example, `gh852`, `gh853`.
  
5. Install dependencies. In the root directory of your local copy, run:

    ```sh
    npm install
    ```

    or (for [Yarn](https://yarnpkg.com/) users):

    ```sh
    yarn
    ```

6. Write code and commit your changes to the branch.

    To build and launch TestCafe Hammerhead, run the `http-playground` task:

    ```sh
    node node_modules/gulp/bin/gulp http-playground
    ```

    The `http-playground` task builds Hammerhead, launches a local HTTP server, and opens a playground page, where you can specify a web page to proxy.

    To run a playground over HTTPS, run the `https-playground` task.

    ```sh

>     node node_modules/gulp/bin/gulp https-playground

Author: Do we need to launch gulp directly from `bin`? Can't we do this instead:

```sh
 node gulp https-playground
 ```

Reviewer 2: You can make it even simpler:

`gulp https-playground`

See step 9 -- there's a short command used.

    ```

7. Add regression tests to appropriate sections if you are fixing a bug. To find these sections, search for `Regression` in the code.

    For new functionality, add unit/functional tests.

8. Fetch upstream changes and rebase your branch onto `master`.

9. Run tests to check that everything works.

    ```sh
    gulp test-server
    gulp test-client
    ```

10. Push changes to your fork and open a pull request.

Before you submit your pull request, it has to satisfy the following conditions:

* The pull request name should describe the changes you implemented.
* The pull request description should contain the [closes](https://github.com/blog/1506-closing-issues-via-pull-requests) directive with an appropriate issue number.

> * All tests must be passing

Reviewer 2: ```suggestion
* All existing and new tests must be passing
```

Probably worth clarifying what tests must pass
